### ISSUE TITLE
[Question] Get details of Inherited instances 

### ISSUE THREAD

**Issue by Anthony-Michel** [json-api-dotnet/JsonApiDotNetCore#1138]
I have a structure with multiple resources inheriting from an abstract resource, serving as main entry point.
It's using EF Core Table-per-hierarchy and discriminator configuration.

a GetAll call give me only the common fields from the abstract resource. 
I need to be able to retrieve the specific fields from each individual child types, and use them for filtering / sorting / selecting fieldsets. 

I tried, as described in the documentation to configure specific Repositories or Services, without success. 
Also saw an MR from 2018 describing how to add a mapping to the configuration, but it seems this logic is now removed. 

Would you get any advice on how to best implement this ?

Some code sample:
EF Configuration: 
```       public void Configure(EntityTypeBuilder<LegalEntity> builder)
        {
            builder.ToTable("LegalEntity");
            builder.HasDiscriminator(e => e.EntityType)
            .HasValue<IndividualEntity>(LegalEntityTypeEnum.Individual.ToString())
            .HasValue<CompanyEntity>(LegalEntityTypeEnum.Company.ToString())
            .HasValue<ListedEntity>(LegalEntityTypeEnum.PubliclyListed.ToString())
            .HasValue<GovernmentEntity>(LegalEntityTypeEnum.Government.ToString());
            ....
 ```
 
 Main Entity: 
 ```    
 public abstract class LegalEntity : Identifiable<long>
    {
        [Attr]
        public string DisplayName { get; set; }
        //discriminator for the different types
        [Attr]
        [Column(TypeName = "varchar(24)")]
        public string EntityType { get; set; }
        ...
```
```
    public class IndividualEntity : LegalEntity
    {
        //basic info
        [Attr]        
        public string Surname { get; set; }
        [Attr]        
        public string Firstname { get; set; }
        ...
```

Controller:
```
    public class LegalEntitiesController : JsonApiQueryController<LegalEntity, long>
    {
        public LegalEntitiesController(IJsonApiOptions options, 
                                    IResourceGraph resourceGraph, 
                                    ILoggerFactory loggerFactory, 
                                    IResourceService<LegalEntity, long> resourceService
                                    ) : base(options, resourceGraph, loggerFactory, resourceService)
        {
        }
    }
```

JsonApi Config
```
            services.AddJsonApi<TenantDbContext>(options =>
            {
                options.Namespace = "api";
                options.UseRelativeLinks = true;
                options.IncludeTotalResourceCount = true;
                options.SerializerOptions.WriteIndented = false;
                options.SerializerOptions.DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull;
                options.SerializerOptions.ReferenceHandler = ReferenceHandler.IgnoreCycles;
                options.SerializerOptions.Converters.Add(new JsonStringEnumConverter());             
#if DEBUG
                options.IncludeExceptionStackTraceInErrors = true;
                options.IncludeRequestBodyInErrors = true;
#endif
            }, discovery => discovery.AddCurrentAssembly());
```

**Comment by bart-degreed** [comment]
Hi @Anthony-Michel, thanks for asking.

Unfortunately, resource inheritance doesn't work very well in JADNC at the moment. It requires quite some work to provide a good and consistent experience, tracked in #844, which is on our [roadmap](https://github.com/json-api-dotnet/JsonApiDotNetCore/blob/master/README.md). While we do have some basic tests [here](https://github.com/json-api-dotnet/JsonApiDotNetCore/tree/master/test/JsonApiDotNetCoreTests/IntegrationTests/ResourceInheritance), #846 contains non-working scenarios we're aware of.

For now, I'd advise to not use inheritance. However, if you want to continue with that, note we welcome PRs with small patches to unblock existing scenarios. Another way to move this forward is to provide feedback and/or suggestions to what's proposed in #844.

**Comment by Anthony-Michel** [comment]
Thank you for the quick answer. 
I really just need it to work on readonly a the moment, even just GetAll :) . 
Based on https://github.com/json-api-dotnet/JsonApiDotNetCore/issues/844 I probably misunderstood this was actually working. 

I don't really see a sustainable way i can get rid of the inheritance for this case to be honest, as the main use case is to be able to let Front end retrieve a list of LegalEntities and their specific details without knowing or filtering/sorting by type. 
While keeping the control on object creation to be able to set only the fields that define the specific type created.

Eventually, for this specific endpoint, i could do a mapping from my EF entities to a flatten version of the resource, as similar to this https://github.com/json-api-dotnet/JsonApiDotNetCore/commit/6097476289b31f572df0dca9deb1e090cdbc6f79
But it seems the code structure. 
Is that still something supported ?

**Comment by bart-degreed** [comment]
Support for entity-resource-separation no longer exists since v4. It was dropped during early development of v4, because there were just too many corner cases that we couldn't get to work.

You could try and remove `abstract` from your base class to see if that unblocks you.

Inheritance is a kind of relationship, so you can always model them to be JSON:API relationships instead. Unless you're willing to contribute fixes to JADNC, that sounds like the best way to move forward at this time.

**Comment by Anthony-Michel** [comment]
Already tried without abstract before posting here :) 
Didn't change much. 
I updated slightly my model meanwhile. 
Looking forward for the capability to be added. Will have a look to see if I can contribute when i'll have some free time.

**Comment by bart-degreed** [comment]
Closing as duplicate of #844.

**Comment by bart-degreed** [comment]
@Anthony-Michel please note the work being done in #1142. If you feel things should work differently, please let me know.

**Comment by Anthony-Michel** [comment]
@bart-degreed thank you for all the work that went into that, greatly appreciated.
It looks exactly like what we need and how i would have do it.
I'll be more than happy to integrate coming version with this change and test it.

**Comment by Anthony-Michel** [comment]
As i was passing by, just realized i forgot to give a feedback, so just for the sake of it, we integrated the new version some time after the last comments and it's working perfectly since.

Once again thank you for all the work that went into it !

**Comment by bkoelman** [comment]
Glad to hear, thanks for the feedback!
